its/src/test/java/com/sonar/it/scanner/msbuild/sonarcloud/IncrementalPRAnalysisSonarCloudTest.java

```diff
@@ -0,0 +1,229 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class IncrementalPRAnalysisSonarCloudTest {
+  private final static Logger LOG = LoggerFactory.getLogger(IncrementalPRAnalysisSonarCloudTest.class);
+  private final static Integer COMMAND_TIMEOUT = 2 * 60 * 1000;
+  private final static String SCANNER_PATH = System.getenv("SCANNER_PATH") == null
+    ? "../build/sonarscanner-msbuild-net46/SonarScanner.MSBuild.exe" // On the local machine, the scanner is prepared by ci-build.ps1 script.
+    : System.getenv("SCANNER_PATH");
+  private final static String[] prArguments = {
+    "/d:sonar.pullrequest.base=master",
+    "/d:sonar.pullrequest.branch=pull-request-branch",
+    "/d:sonar.pullrequest.key=pull-request-key"
+  };
+  private final static String SONARCLOUD_ORGANIZATION = System.getenv("SONARCLOUD_ORGANIZATION");
+  private final static String SONARCLOUD_PROJECT_KEY = System.getenv("SONARCLOUD_PROJECT_KEY");
+  private final static String SONARCLOUD_URL = System.getenv("SONARCLOUD_URL");
+  private final static String SONARCLOUD_PROJECT_TOKEN = System.getenv("SONARCLOUD_PROJECT_TOKEN");
+
+  @ClassRule
+  public static TemporaryFolder temp = TestUtils.createTempFolder();
+
+  @Test
+  public void incrementalPrAnalysis_master() throws IOException {
+    var projectDir = TestUtils.projectDir(temp, "IncrementalPRAnalysis");
+    var logWriter = new StringWriter();
+    StreamConsumer.Pipe logsConsumer = new StreamConsumer.Pipe(logWriter);
+
+    runBeginStep(projectDir, logsConsumer);
+
+    assertThat(logWriter.toString()).contains(
+      "Processing analysis cache",
+      "Incremental PR analysis: Base branch parameter was not provided.",
+      "Cache data is empty. A full analysis will be performed.");
+  }
+
+  @Test
+  public void incrementalPrAnalysis_prWithoutChanges_producesUnchangedFilesWithAllFiles() throws IOException {
+    var projectDir = TestUtils.projectDir(temp, "IncrementalPRAnalysis");
+
+    runAnalysis(projectDir); // Initial build - master.
+    var logs = runAnalysis(projectDir, prArguments); // PR analysis.
+
+    // Verify that the file hashes are considered and all of them will be skipped.
+    assertThat(logs).contains("Incremental PR analysis: 3 files out of 3 are unchanged.");
+    Path unchangedFilesPath = getUnchangedFilesPath(projectDir);
+    assertThat(Files.readString(unchangedFilesPath)).contains("Unchanged1.cs", "Unchanged2.cs", "WithChanges.cs");
+  }
+
+  @Test
+  public void incrementalPrAnalysis_prWithChanges() throws IOException {
+    var projectDir = TestUtils.projectDir(temp, "IncrementalPRAnalysis");
+
+    runAnalysis(projectDir); // Initial build - master.
+    changeFile(projectDir, "IncrementalPRAnalysis\\WithChanges.cs"); // Change a file to force analysis.
+    runAnalysis(projectDir, prArguments); // PR analysis.
+
+    // Assert that `WithChanges.cs` file is considered modified and will be analyzed.
```

Comment:
Extremely pedantic, but it took me a couple of seconds to read this correctly:
"will be analyzed", should this not be "`was` analyzed"?
Since the full analysis has already been executed.